src/box/tuple_format.c

```diff
@@ -81,6 +81,11 @@ tuple_format_cmp(const struct tuple_format *format1,
 {
 	struct tuple_format *a = (struct tuple_format *)format1;
 	struct tuple_format *b = (struct tuple_format *)format2;
+	/*
+	 * FIXME(gh-8099): We cannot compare MsgPack data of formats, since
+	 * formats of ephemeral spaces don't have MsgPack data.
+	 */
+
```

Comment:
That isn't the only reason - your current binary format is user-provided. Each field is a map of key-value pairs. Maps are not ordered in msgpack. So the same binary format could be stored as `name = name1, type = type1` or `type = type1, name = name1`. Direct `memcmp()` would see them as different formats. The solution would be not to allow users to push stuff here directly. Instead, you could decode+validate it, and then encode back it back in `tuple_format_new()` with a predictable order and in a compact form.

Again, I am not insisting on anything. Putting it here for the record.

Reply:
> That isn't the only reason - your current binary format is user-provided. Each field is a map of key-value pairs. Maps are not ordered in msgpack. So the same binary format could be stored as `name = name1, type = type1` or `type = type1, name = name1`.

We thought about this, but considered this not to be a relevant case.